NBitcoin/Protocol/NodeSocketEventManager.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+
+namespace NBitcoin.Protocol
+{
+	/// <summary>
+	/// Singleton class which deals with a single instance of <see cref="SocketAsyncEventArgs"/>.
+	/// </summary>
+	internal sealed class NodeSocketEventManager : IDisposable
+	{
+		private SocketAsyncEventArgs instance;
+
+		private NodeSocketEventManager() { }
```

Comment:
Do we need this?

Reply:
Yes, so that we can't instantiate the Manager without going through the static Create() method.